asmcli/lib/util.sh

```diff
@@ -22,3 +22,978 @@ gen_install_params() {
+        context_set-option "CUSTOM_CA" 1
+        shift 2
+        ;;
+      -r | --revision_name | --revision-name)
+        arg_required "${@}"
+        context_set-option "CUSTOM_REVISION" 1
+        REVISION_LABEL="${2}"
+        shift 2
+        ;;
+      --platform)
+        arg_required "${@}"
+        context_set-option "PLATFORM" "$(echo "${2}" | tr '[:upper:]' '[:lower:]')"
+        shift 2
+        ;;
+      -v | --verbose)
+        context_set-option "VERBOSE" 1
+        shift 1
+        ;;
+      -h | --help)
+        context_set-option "PRINT_HELP" 1
+        shift 1
+        ;;
+      --version)
+        context_set-option "PRINT_VERSION" 1
+        shift 1
+        ;;
+      *)
+        fatal_with_usage "Unknown option ${1}"
+        ;;
+    esac
+  done
+  readonly REVISION_LABEL
+
+  local PRINT_HELP; PRINT_HELP="$(context_get-option "PRINT_HELP")"
+  local PRINT_VERSION; PRINT_VERSION="$(context_get-option "PRINT_VERSION")"
+  local VERBOSE; VERBOSE="$(context_get-option "VERBOSE")"
+  if [[ "${PRINT_HELP}" -eq 1 || "${PRINT_VERSION}" -eq 1 ]]; then
+    if [[ "${PRINT_VERSION}" -eq 1 ]]; then
+      version_message
+    elif [[ "${VERBOSE}" -eq 1 ]]; then
+      usage
+    else
+      usage_short
+    fi
+    exit
+  fi
+}
+
+download_kpt() {
+  local PLATFORM
+  PLATFORM="$(get_platform "$(uname)" "$(uname -m)")"
+  if [[ -z "${PLATFORM}" ]]; then
+    fatal "$(uname) $(uname -m) is not a supported platform to perform installation from."
+  fi
+
+  local KPT_TGZ
+  KPT_TGZ="https://github.com/GoogleContainerTools/kpt/releases/download/v0.39.3/kpt_${PLATFORM}-0.39.3.tar.gz"
+
+  info "Downloading kpt.."
+  curl -L "${KPT_TGZ}" | tar xz
+  AKPT="$(apath -f kpt)"
+}
+
+get_platform() {
+  local OS; OS="${1}"
+  local ARCH; ARCH="${2}"
+  local PLATFORM
+
+  case "${OS}" in
+    Linux ) PLATFORM="linux_amd64";;
+    Darwin)
+      if [[ "${ARCH}" == "arm64" ]]; then
+        PLATFORM="darwin_arm64"
+      else
+        PLATFORM="darwin_amd64"
+      fi
+    ;;
+  esac
+
+  echo "${PLATFORM}"
+}
+
+download_asm() {
+  local OS
+  local PROJECT_ID; PROJECT_ID="$(context_get-option "PROJECT_ID")"
+
+  case "$(uname)" in
+    Linux ) OS="linux-amd64";;
+    Darwin) OS="osx";;
+    *     ) fatal "$(uname) is not a supported OS.";;
+  esac
+
+  info "Downloading ASM.."
+  local TARBALL; TARBALL="istio-${RELEASE}-${OS}.tar.gz"
+  if [[ -z "${_CI_ASM_PKG_LOCATION}" ]]; then
+    curl -L "https://storage.googleapis.com/gke-release/asm/${TARBALL}" \
+      | tar xz
+  else
+    local TOKEN; TOKEN="$(retry 2 gcloud --project="${PROJECT_ID}" auth print-access-token)"
+    run_command curl -L "https://storage.googleapis.com/${_CI_ASM_PKG_LOCATION}/asm/${TARBALL}" \
+      --header @- <<EOF | tar xz
+Authorization: Bearer ${TOKEN}
+EOF
+  fi
+
+  ln -s "${ISTIOCTL_REL_PATH}" .
+
+  info "Downloading ASM kpt package..."
+  retry 3 kpt pkg get --auto-set=false "${KPT_URL}" asm
+  version_message > "${ASM_VERSION_FILE}"
+}
+
+#######
```

Comment:
I think from here down to `get_auth_token` would be good candidates for combining with the stuff in `iam.sh` and calling it "installation_dependencies.sh" or something.

Reply:
Done.